overlays/tools.nix

```diff
@@ -67,6 +86,16 @@ in { haskell-nix = prev.haskell-nix // {
 
   tools = lib.mapAttrs final.haskell-nix.tool;
 
+  # Like `tool` but allows default ghc to be specified by nix name
+  tool' = compiler-nix-name: name: versionOrArgs:
```

Comment:
Isn't the whole point of `tool` that it uses the same GHC as your project? So if you're not using that... why not just use `hackage-package`? Each of these functions seems to cost us a lot of extra code now, so I'm keen to have as few as possible...

Reply:
The `tool'` and `tools'` functions are used to implement `p.tool` and `p.tools`.  I am not sure if we should encourage anyone to use them, but:

```
tool' "ghc865" "hello" "1.0.0.2"
```
is a lot more concise than
```
(hackage-package { name = "hello"; version = "1.0.0.2"; compiler-nix-name = "ghc865" }).components.exes.hello
```